src/migration/migration-rules.js

```diff
@@ -1605,6 +1605,80 @@ module.exports = {
         delete north.authentication.password
         delete north.authentication.token
       }
+
+      const northCache = path.resolve('./cache/data-stream', `north-${north.id}`)
+      if (await filesExists(path.resolve(northCache, 'errors'))) {
+        logger.info('Rename north cache errors to files-errors')
+        try {
+          await fs.rename(path.resolve(northCache, 'errors'), path.resolve(northCache, 'files-errors'))
+        } catch (error) {
+          logger.error(error)
+        }
+      }
+
+      try {
+        await createFolder(path.resolve(northCache, 'values'))
+        await fs.writeFile(
+          path.resolve(northCache, 'values', 'buffer.tmp'),
+          '[]',
+          { encoding: 'utf8', flag: 'w' },
+        )
+      } catch (writeBufferError) {
+        logger.error(writeBufferError)
+      }
+
+      if (await filesExists(path.resolve(northCache, 'values.db'))) {
+        logger.info(`Migrating values.db for North connector ${north.id}`)
+        const database = db(path.resolve(northCache, 'values.db'))
+        const query = 'SELECT * '
+            + 'FROM cache '
+            + 'LIMIT ?'
+        let results = []
+        do {
+          try {
+            results = database.prepare(query).all(north.caching.maxSendCount)
+          } catch (error) {
+            logger.error(error)
+          }
+
+          if (results.length > 0) {
+            try {
+              const compactFileName = `${nanoid()}.compact.tmp`
+              await fs.writeFile(
+                path.resolve(northCache, 'values', compactFileName),
+                JSON.stringify(results.map((value) => ({
+                  timestamp: value.timestamp,
+                  pointId: value.pointId,
+                  data: JSON.parse(decodeURI(value.data)),
+                }))),
+                { encoding: 'utf8', flag: 'w' },
+              )
+            } catch (writeError) {
+              if (writeError.code !== 'ENOENT') {
+                logger.error(writeError)
+              }
+            }
+
+            try {
+              const ids = results.map((value) => value.id).join()
+              const removeQuery = `DELETE FROM cache WHERE id IN (${ids})`
+              const result = database.prepare(removeQuery).run()
+              logger.info(`${result.changes} values migrated for North connector ${north.name} (${north.id}).`)
+            } catch (error) {
+              logger.error(error)
+            }
+          }
+        } while (results.length > 0)
+
+        try {
+          await fs.rm(path.resolve(`./cache/data-stream/north-${north.id}/values.db`), { force: true })
+          await fs.rm(path.resolve(`./cache/data-stream/north-${north.id}/values-error.db`), { force: true })
+        } catch (err) {
+          if (err.code !== 'ENOENT') {
+            logger.error(err)
+          }
+        }
+      }
```

Comment:
should we delete the file only if the migration succeeded? with all try catch i am not sure what is done finally

Reply:
If the database is migrated, then the database for this specific north can be removed. In case of removal error I think the migration should go on with other connectors instead of blocking. I add a code comment to explain that